app/src/main/java/org/mozilla/fenix/utils/WifiConnectionMonitor.kt

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.utils
+
+import android.app.Application
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+
+/**
+ * Attaches itself to the [Application] and listens for WIFI available/not available events. This
+ * allows calling code to set simpler listeners.
+ *
+ * Example:
+ * ```kotlin
+ *  app.components.wifiConnectionListener.addOnWifiConnectedChangedListener { isConnected ->
+ *    if (isConnected) {
+ *      downloadThing()
+ *    }
+ *  }
+ * ```
+ */
+class WifiConnectionMonitor(app: Application) {
```

Comment:
This is a good candidate to be completely ported  to AC :)

Reply:
Didn't occur to me until now, and I think it's out of scope for this story, but we should change how this class exposes its callbacks if/when we move it to AC.  We would get two big benefits if, instead of just adding/removing callbacks, we exposed this as a `WifiStore` (with `WifiState(val isConnected: Boolean)`).  1 if we ever needed to listen to other framework wifi callbacks, adding them would be simple and easy (as is it would get messy fast), and 2 it would be much more in line with the rest of our architecture.

Reply:
👍